dev_requirements.txt

```diff
@@ -1,19 +1,14 @@
-# third-party integration
-pyquil~=2.18.0
-qiskit~=0.23.1
+# Third-party integration.
+qiskit
+pyquil
```

Comment:
I am not sure removing the pinning is good.
See this issue about a similar proposal: https://github.com/unitaryfund/mitiq/issues/508

Reply:
The problem with pinning is when we update one pin and not others pip can break, so I'm not sure I agree we should pin these. Wouldn't we want to be up to date with the latest qiskit/pyquil versions anyway? For tools like mypy I agree we should pin.

Reply:
I would also be skeptical of removing pinning here as other packages can push breaking changes that would then break our stuff forcing a patch (if we found out).